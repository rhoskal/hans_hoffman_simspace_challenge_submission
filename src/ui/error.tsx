import React from "react";

type Props = {
  message?: string;
};

export function Error({ message = "Ah ah ah, you didn't say the magic word." }: Props) {
  return <p>{message}</p>;
}
