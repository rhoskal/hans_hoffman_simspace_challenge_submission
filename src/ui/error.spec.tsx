import React from "react";
import { render } from "@testing-library/react";

import { Error } from "./error";

describe("[Component]: Error", function() {
  it("should allow custom error message", function() {
    const msg: string = "No breeds were found.";
    const { getByText } = render(<Error message={msg} />);

    getByText(msg);
  });

  it("should match snapshot", function() {
    const { container } = render(<Error />);

    expect(container).toMatchSnapshot();
  });
});
