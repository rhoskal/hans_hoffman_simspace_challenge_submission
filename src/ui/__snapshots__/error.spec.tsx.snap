// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Error should match snapshot 1`] = `
<div>
  <p>
    Ah ah ah, you didn't say the magic word.
  </p>
</div>
`;
